snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedMemberOfFieldFixRequest.java

```diff
@@ -0,0 +1,190 @@
+			memberOfDescriptionIterator.forEachRemaining(descriptions -> descriptions.forEach(description -> missingMembersOf.removeAll(description.getId())));
+			
+			missingMembersOfComponents.putAll(missingMembersOf);
+			log.info("Found {} components with missing member of entry for reference set {}", missingMembersOf.size(), referenceSetId);
+			
+			// Find concepts/descriptions with incomplete active member of fields
+			SnomedConceptSearchRequestBuilder conceptActiveMemberOfRequest = SnomedRequests.prepareSearchConcept().isActiveMemberOf(referenceSetId).setFields(ID).setLimit(50_000);
+			SearchResourceRequestIterator<SnomedConceptSearchRequestBuilder, SnomedConcepts> activeMemberOfConceptIterator = 
+					new SearchResourceRequestIterator<>(conceptActiveMemberOfRequest, b -> b.build().execute(context));
+			activeMemberOfConceptIterator.forEachRemaining(concepts -> concepts.forEach(concept -> missingActiveMembersOf.removeAll(concept.getId())));
+	
+			SnomedDescriptionSearchRequestBuilder descriptionActiveMemberOfRequest = SnomedRequests.prepareSearchDescription().isActiveMemberOf(referenceSetId).setFields(ID).setLimit(50_000);
+			SearchResourceRequestIterator<SnomedDescriptionSearchRequestBuilder, SnomedDescriptions> activeMemberOfDescriptionIterator = 
+					new SearchResourceRequestIterator<>(descriptionActiveMemberOfRequest, b -> b.build().execute(context));
+			activeMemberOfDescriptionIterator.forEachRemaining(descriptions -> descriptions.forEach(description -> missingActiveMembersOf.removeAll(description.getId())));
+			
+			missingActiveMembersOfComponents.putAll(missingActiveMembersOf);
+			log.info("Found {} components with missing active member of entry for reference set {}",  missingActiveMembersOf.size(), referenceSetId);
+
+		}
+		
+		int modifiedComponentCount = 0;
+		
+		Set<String> allAffectedComponents = Sets.union(missingMembersOfComponents.keySet(), missingActiveMembersOfComponents.keySet());
+		
+		for (String componentId : allAffectedComponents) {
+			try {
+				
+				if (SnomedIdentifiers.isConceptIdentifier(componentId)) {
+					
+					final SnomedConceptDocument concept = context.lookup(componentId, SnomedConceptDocument.class);				
+					List<String> memberOf = new ArrayList<>(concept.getMemberOf());
+					if (missingMembersOfComponents.containsKey(componentId)) {
+						memberOf.addAll(missingMembersOfComponents.get(componentId));				
+					}
+					
+					List<String> activeMemberOf = new ArrayList<>(concept.getActiveMemberOf());
+					if (missingActiveMembersOfComponents.containsKey(componentId)) {
+						activeMemberOf.addAll(missingActiveMembersOfComponents.get(componentId));
+					}
+					
+					SnomedConceptDocument updatedDocument = SnomedConceptDocument.builder(concept)
+							.memberOf(memberOf)
+							.activeMemberOf(activeMemberOf)
+							.build();
+					context.update(concept, updatedDocument);
+					
+				} else {
```

Comment:
I would do a check here to ensure that we are handling a description and log everything else in another else branch. This also means that the modifiedComponentCount should be moved inside the if-else block (increment for concepts and descriptions)
The likelihood is low but who knows?